android/uhabits-android/src/main/java/org/isoron/uhabits/widgets/views/CheckmarkWidgetView.java

```diff
@@ -49,74 +53,76 @@
 
     protected int checkmarkValue;
 
-    protected int checkmarkState;
+    protected CheckmarkState checkmarkState;
 
     protected boolean isNumerical;
 
-    public CheckmarkWidgetView(Context context)
-    {
+
+    public CheckmarkWidgetView(Context context) {
         super(context);
         init();
     }
 
-    public CheckmarkWidgetView(Context context, AttributeSet attrs)
-    {
+
+    public CheckmarkWidgetView(Context context, AttributeSet attrs) {
         super(context, attrs);
         init();
     }
 
-    public void refresh()
-    {
-        if (backgroundPaint == null || frame == null || ring == null) return;
+
+    public void refresh() {
+        if (backgroundPaint == null || frame == null || ring == null) {
+            return;
+        }
 
         StyledResources res = new StyledResources(getContext());
 
         int bgColor;
         int fgColor;
 
-        switch (checkmarkState) {
-            case Checkmark.YES_MANUAL:
-            case Checkmark.SKIP:
-                bgColor = activeColor;
-                fgColor = res.getColor(R.attr.highContrastReverseTextColor);
-                setShadowAlpha(0x4f);
-                backgroundPaint.setColor(bgColor);
-                frame.setBackgroundDrawable(background);
-                break;
-
-            case Checkmark.YES_AUTO:
-            case Checkmark.NO:
-            default:
-                getResources().getString(R.string.fa_times);
-                bgColor = res.getColor(R.attr.cardBgColor);
-                fgColor = res.getColor(R.attr.mediumContrastTextColor);
-                setShadowAlpha(0x00);
-                break;
+        if (checkmarkState.isManualInput()) {
```

Comment:
I'm not 100% sure whether this is the desired behaviour. So this needs to be clarified.